yb-voyager/cmd/assessMigrationCommand.go

```diff
@@ -422,76 +409,107 @@ func createMigrationAssessmentCompletedEvent() *cp.MigrationAssessmentCompletedE
+		AssessmentIssues:    assessmentIssues,
 		SourceSizeDetails: SourceDBSizeDetails{
-			TotalIndexSize:     sizeDetails.TotalIndexSize,
-			TotalTableSize:     sizeDetails.TotalTableSize,
-			TotalTableRowCount: sizeDetails.TotalTableRowCount,
+			TotalIndexSize:     assessmentReport.GetTotalIndexSize(),
+			TotalTableSize:     assessmentReport.GetTotalTableSize(),
+			TotalTableRowCount: assessmentReport.GetTotalTableRowCount(),
 			TotalDBSize:        source.DBSize,
 		},
 		TargetRecommendations: TargetSizingRecommendations{
-			TotalColocatedSize: sizeDetails.TotalColocatedSize,
-			TotalShardedSize:   sizeDetails.TotalShardedSize,
+			TotalColocatedSize: totalColocatedSize,
+			TotalShardedSize:   totalShardedSize,
 		},
-		MigrationComplexity: assessmentReport.MigrationComplexity,
-		ConversionIssues:    schemaAnalysisReport.Issues,
+		ConversionIssues:     schemaAnalysisReport.Issues,
+		AssessmentJsonReport: assessmentReport,
 	}
 
-	finalReportBytes, err := json.Marshal(finalReport)
+	payloadBytes, err := json.Marshal(payload)
 	if err != nil {
 		utils.PrintAndLog("Failed to serialise the final report to json (ERR IGNORED): %s", err)
 	}
 
-	ev.Report = string(finalReportBytes)
+	ev.Report = string(payloadBytes)
+	log.Infof("assess migration payload send to yugabyted: %s", ev.Report)
 	return ev
 }
 
-type SizeDetails struct {
-	TotalIndexSize     int64
-	TotalTableSize     int64
-	TotalTableRowCount int64
-	TotalColocatedSize int64
-	TotalShardedSize   int64
-}
+// flatten UnsupportedDataTypes, UnsupportedFeatures, MigrationCaveats
+func flattenAssessmentReportToAssessmentIssues(ar AssessmentReport) []AssessmentIssuePayload {
+	var issues []AssessmentIssuePayload
 
-func (ar *AssessmentReport) CalculateSizeDetails(dbType string) (SizeDetails, error) {
-	var details SizeDetails
-	colocatedTables, err := ar.GetColocatedTablesRecommendation()
-	if err != nil {
-		return details, fmt.Errorf("failed to get the colocated tables recommendation: %v", err)
+	var docsLink string
+	switch source.DBType {
+	case POSTGRESQL:
+		docsLink = UNSUPPORTED_DATATYPES_DOC_LINK
+	case ORACLE:
+		docsLink = UNSUPPORTED_DATATYPES_DOC_LINK_ORACLE
+	}
+	for _, unsupportedDataType := range ar.UnsupportedDataTypes {
+		issues = append(issues, AssessmentIssuePayload{
+			Type:            DATATYPE,
+			TypeDescription: DATATYPE_ISSUE_TYPE_DESCRIPTION,
+			Subtype:         unsupportedDataType.DataType,
+			ObjectName:      fmt.Sprintf("%s.%s.%s", unsupportedDataType.SchemaName, unsupportedDataType.TableName, unsupportedDataType.ColumnName),
+			SqlStatement:    "",
+			DocsLink:        docsLink,
+		})
+	}
+
+	for _, unsupportedFeature := range ar.UnsupportedFeatures {
+		for _, object := range unsupportedFeature.Objects {
+			issues = append(issues, AssessmentIssuePayload{
+				Type:               FEATURE,
+				TypeDescription:    FEATURE_ISSUE_TYPE_DESCRIPTION,
+				Subtype:            unsupportedFeature.FeatureName,
+				SubtypeDescription: "", // TODO
```

Comment:
you can directly use the `unsupportedFeature.FeatureDescription` for now, whenever we start adding it will automatically come up.